File: flyrxbus/src/main/java/com/xia/flyrxbus/RxBusManager.java
package com.xia.flyrxbus;

import io.reactivex.Scheduler;

@SuppressWarnings({"unused", "WeakerAccess"})
public final class RxBusManager {

    public static void post(final Object event) {
        RxBus.getDefault().post(event);
    }

    public static void post(final Object event, final String tag) {
        RxBus.getDefault().post(event, tag);
    }

    public static void postSticky(final Object event) {
        RxBus.getDefault().postSticky(event);
    }

    public static void postSticky(final Object event, final String tag) {
        RxBus.getDefault().postSticky(event, tag);
    }

    public static <T> void subscribe(final Object subscriber,
                                     final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribe(subscriber, callback);
    }

    public static <T> void subscribe(final Object subscriber,
                                     final Scheduler scheduler,
                                     final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribe(subscriber, scheduler, callback);
    }

    public static <T> void subscribe(final Object subscriber,
                                     final String tag,
                                     final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribe(subscriber, tag, callback);
    }

    public static <T> void subscribe(final Object subscriber,
                                     final String tag,
                                     final Scheduler scheduler,
                                     final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribe(subscriber, tag, scheduler, callback);
    }

    public static <T> void subscribeSticky(final Object subscriber,
                                           final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribeSticky(subscriber, callback);
    }

    public static <T> void subscribeSticky(final Object subscriber,
                                           final Scheduler scheduler,
                                           final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribeSticky(subscriber, scheduler, callback);
    }

    public static <T> void subscribeSticky(final Object subscriber,
                                           final String tag,
                                           final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribeSticky(subscriber, tag, callback);
    }

    public static <T> void subscribeSticky(final Object subscriber,
                                           final String tag,
                                           final Scheduler scheduler,
                                           final RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribeSticky(subscriber, tag, scheduler, callback);
    }

    public static <T> void subscribeWithTags(final Object subscriber,
                                             final RxBus.Callback<T> callback,
                                             final String... tags) {
        subscribeWithTags(subscriber, null, callback, tags);
    }

    public static <T> void subscribeWithTags(final Object subscriber,
                                             final Scheduler scheduler,
                                             final RxBus.Callback<T> callback,
                                             final String... tags) {
        if (tags == null || tags.length == 0) {
            throw new RuntimeException("Tags is empty,you should set the tags");
        }
        for (String tag : tags) {
            RxBus.getDefault().subscribe(subscriber, tag, scheduler, callback);
        }
    }

    public static <T> void subscribeStickyWithTags(final Object subscriber,
                                                   final RxBus.Callback<T> callback,
                                                   final String... tags) {
        subscribeStickyWithTags(subscriber, null, callback, tags);
    }

    public static <T> void subscribeStickyWithTags(final Object subscriber,
                                                   final Scheduler scheduler,
                                                   final RxBus.Callback<T> callback,
                                                   final String... tags) {
        if (tags == null || tags.length == 0) {
            throw new RuntimeException("Tags is empty,you should set the tags");
        }
        for (String tag : tags) {
            RxBus.getDefault().subscribeSticky(subscriber, tag, scheduler, callback);
        }
    }

    public static void unregister(final Object subscriber) {
        RxBus.getDefault().unregister(subscriber);
    }
}

File: flyrxbus/src/main/java/com/xia/flyrxbus/RxBusManager.kt
package com.xia.flyrxbus

import androidx.annotation.NonNull
import io.reactivex.Scheduler

@Suppress("unused")
object RxBusManager {

    @JvmStatic
    fun post(@NonNull event: Any) {
        RxBus.post(event)
    }

    @JvmStatic
    fun post(@NonNull event: Any, @NonNull tag: String) {
        RxBus.post(event, tag)
    }

    @JvmStatic
    fun postSticky(@NonNull event: Any) {
        RxBus.postSticky(event)
    }

    @JvmStatic
    fun postSticky(@NonNull event: Any, @NonNull tag: String) {
        RxBus.postSticky(event, tag)
    }

    @JvmStatic
    fun removeSticky(@NonNull event: Any) {
        RxBus.removeSticky(event)
    }

    @JvmStatic
    fun removeSticky(@NonNull event: Any,
                     @NonNull tag: String) {
        RxBus.removeSticky(event, tag)
    }

    @JvmStatic
    fun <T> subscribe(@NonNull subscriber: Any,
                      @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribe(subscriber, callback)
    }

    @JvmStatic
    fun <T> subscribe(@NonNull subscriber: Any,
                      scheduler: Scheduler?,
                      @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribe(subscriber, scheduler, callback)
    }

    @JvmStatic
    fun <T> subscribe(@NonNull subscriber: Any,
                      @NonNull tag: String,
                      @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribe(subscriber, tag, callback)
    }

    @JvmStatic
    fun <T> subscribe(@NonNull subscriber: Any,
                      @NonNull tag: String,
                      scheduler: Scheduler?,
                      @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribe(subscriber, tag, scheduler, callback)
    }

    @JvmStatic
    fun <T> subscribeSticky(@NonNull subscriber: Any,
                            @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribeSticky(subscriber, callback)
    }

    @JvmStatic
    fun <T> subscribeSticky(@NonNull subscriber: Any,
                            scheduler: Scheduler?,
                            @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribeSticky(subscriber, scheduler, callback)
    }

    @JvmStatic
    fun <T> subscribeSticky(@NonNull subscriber: Any,
                            @NonNull tag: String,
                            @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribeSticky(subscriber, tag, callback)
    }

    @JvmStatic
    fun <T> subscribeSticky(@NonNull subscriber: Any,
                            @NonNull tag: String,
                            scheduler: Scheduler?,
                            @NonNull callback: RxBus.Callback<T>) {
        RxBus.subscribeSticky(subscriber, tag, scheduler, callback)
    }

    @JvmStatic
    fun <T> subscribeWithTags(@NonNull subscriber: Any,
                              @NonNull callback: RxBus.Callback<T>,
                              @NonNull vararg tags: String) {
        subscribeWithTags(subscriber, null, callback, *tags)
    }

    @JvmStatic
    fun <T> subscribeWithTags(@NonNull subscriber: Any,
                              scheduler: Scheduler?,
                              @NonNull callback: RxBus.Callback<T>,
                              @NonNull vararg tags: String) {
        if (tags.isEmpty()) {
            throw RuntimeException("Tags is empty,you should set the tags")
        }
        for (tag in tags) {
            RxBus.subscribe(subscriber, tag, scheduler, callback)
        }
    }

    @JvmStatic
    fun <T> subscribeStickyWithTags(@NonNull subscriber: Any,
                                    @NonNull callback: RxBus.Callback<T>,
                                    @NonNull vararg tags: String) {
        subscribeStickyWithTags(subscriber, null, callback, *tags)
    }

    @JvmStatic
    fun <T> subscribeStickyWithTags(@NonNull subscriber: Any,
                                    scheduler: Scheduler?,
                                    @NonNull callback: RxBus.Callback<T>,
                                    @NonNull vararg tags: String) {
        if (tags.isEmpty()) {
            throw RuntimeException("Tags is empty,you should set the tags")
        }
        for (tag in tags) {
            RxBus.subscribeSticky(subscriber, tag, scheduler, callback)
        }
    }

    @JvmStatic
    fun unregister(@NonNull subscriber: Any) {
        RxBus.unregister(subscriber)
    }
}
